components/brave_rewards/common/publisher_utils.cc

```diff
@@ -1,33 +1,48 @@
 /* Copyright (c) 2022 The Brave Authors. All rights reserved.
  * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-#include "brave/components/brave_rewards/browser/publisher_utils.h"
+#include "brave/components/brave_rewards/common/publisher_utils.h"
 
 #include <array>
 #include <optional>
 
 #include "base/ranges/algorithm.h"
+#include "base/strings/string_util.h"
 #include "net/base/registry_controlled_domains/registry_controlled_domain.h"
 
 namespace brave_rewards {
 
 namespace {
 
-const std::array kMediaPlatformDomains = {"twitter.com", "github.com",
-                                          "reddit.com",  "twitch.tv",
-                                          "vimeo.com",   "youtube.com"};
+constexpr auto kMediaPlatformDomains = std::to_array<std::string_view>(
+    {"github.com", "reddit.com", "twitch.tv", "twitter.com", "vimeo.com",
+     "x.com", "youtube.com"});
+
+constexpr auto kMediaPlatformPrefixes = std::to_array<std::string_view>(
+    {"github#", "reddit#", "twitch#", "twitter#", "vimeo#", "youtube#"});
+
+}  // namespace
 
 bool IsMediaPlatformURL(const GURL& url) {
-  return base::ranges::any_of(kMediaPlatformDomains, [&url](auto* domain) {
+  return base::ranges::any_of(kMediaPlatformDomains, [&url](auto domain) {
     return net::registry_controlled_domains::SameDomainOrHost(
```

Comment:
can we just do an exact domain match for one of the domains in https://github.com/brave/brave-core/pull/25394/files#diff-ae0f5363bd7d8495d043fe54f0af1637334c1711d0ae52de6517ebc6e36e6334R32 (kScriptMap)? that way we don't match unnecessary subdomains of these domains

Reply:
For this function, we do want to match subdomains, so that we don't end up treating a subdomain of a "media site" as a regular site for AC tracking purposes.

For instance, we don't want user activity on `something.x.com` to be recorded as attention on `x.com`.